### PR TITLE
Support Webpack 4 plugin system

### PR DESCRIPTION
**Pull request by oscargullberg**
Fixes https://github.com/webpack-contrib/zopfli-webpack-plugin/issues/11

Further development of https://github.com/webpack-contrib/zopfli-webpack-plugin/pull/12

### CODE REVIEW

**Review by alexander-akait — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -88,7 +88,13 @@ class ZopfliPlugin {
           cb();
         });
       }, callback);
-    });
+    };
+
+    if (compiler.hooks) {
+      compiler.hooks.emit.tapAsync(this.constructor.name, processAssets);
```
Comment (alexander-akait) [inline src/index.js]: Oh, sorry, please use `ZopfliPlugin` string instead `this.constructor.name`

**Review by alexander-akait — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (oscargullberg) [reply src/index.js]: Fixed

**Review by alexander-akait — CHANGES_REQUESTED** [review]
Can you fix lint problems https://travis-ci.org/webpack-contrib/zopfli-webpack-plugin/jobs/383068815 in this PR and we can merge?

**Comment by alexander-akait** [comment]
@oscargullberg friendly ping

**Review by alexander-akait — APPROVED** [review]

**Comment by balloob** [comment]
Just submitted a PR to @oscargullberg to resolve the remaining linting error https://github.com/oscargullberg/zopfli-webpack-plugin/pull/1

**Comment by balloob** [comment]
`nsp` fails because of a high and two medium alerts for `node-zopfli`. So if we combine this PR with #15, it should get those fixed.

**Comment by alexander-akait** [comment]
This package deprecated in favor https://github.com/webpack-contrib/compression-webpack-plugin, please look how use `zopfli` library https://github.com/webpack-contrib/compression-webpack-plugin#examples. You can use `node-zopfli` instead `@gfx/zopfli` if you need `node@6` support. Thanks for using webpack.

**Pull request closed without merging**